resources/repo.rb

```diff
@@ -2,17 +2,18 @@
 property :repo_type, String, desired_state: false, identity: true, default: 'maven2-hosted'.freeze
 property :attributes, Hash, default: lazy { ::Mash.new } # Not mandatory but strongly recommended in the generic case.
 property :online, [true, false], default: true
-property :api_url, String, desired_state: false, identity: true, default: 'http://localhost:8081/service/siesta/rest/v1/script/'.freeze
-property :api_user, String, desired_state: false, identity: true, default: 'admin'.freeze
-property :api_password, String, desired_state: false, identity: true, sensitive: true, default: 'admin123'.freeze
+property :api_endpoint, String, desired_state: false, identity: true, default: node['nexus3']['api']['endpoint']
+property :api_username, String, desired_state: false, identity: true, default: node['nexus3']['api']['username']
+property :api_password, String, desired_state: false, identity: true, sensitive: true,
+                                default: node['nexus3']['api']['password']
 
 def apiclient
-  @apiclient ||= ::Nexus3::Api.new(api_url, api_user, api_password)
+  @apiclient ||= ::Nexus3::Api.new(api_endpoint, api_username, api_password)
```

Comment:
new_resource must be put in place

Reply:
done and moved apiclient method inside load_current_value